dist/dialog-polyfill.esm.js

```diff
@@ -91,6 +96,48 @@ function isFormMethodDialog(el) {
   return el.getAttribute('method').toLowerCase() === 'dialog';
 }
 
+/**
+ * @param {!DocumentFragment|!Element} hostElement
+ * @return {?Element}
+ */
+function findFocusableElementWithin(hostElement) {
+  // Note that this is 'any focusable area'. This list is probably not exhaustive, but the
+  // alternative involves stepping through and trying to focus everything.
+  var opts = ['button', 'input', 'keygen', 'select', 'textarea'];
+  var query = opts.map(function(el) {
+    return el + ':not([disabled])';
+  });
+  // TODO(samthor): tabindex values that are not numeric are not focusable.
+  query.push('[tabindex]:not([disabled]):not([tabindex=""])');  // tabindex != "", not disabled
+  var target = hostElement.querySelector(query.join(', '));
+
+  if (!target && 'attachShadow' in Element.prototype) {
+    // If we haven't found a focusable target, see if the host element contains an element
+    // which has a shadowRoot.
+    // Recursively search for the first focusable item in shadow roots.
+    var elems = hostElement.querySelectorAll('*');
+    for (var i = 0; i < elems.length; i++) {
+      if (elems[i].tagName && elems[i].shadowRoot) {
+        target = findFocusableElementWithin(elems[i].shadowRoot);
+        if (target) {
+          break;
+        }
+      }
+    }
+  }
+  return target;
+}
+
+/**
+ * Determines if an element is attached to the DOM.
+ * @param {Element} element to check
+ * @return {Boolean} whether the element is in DOM
+ */
+function isConnected(element) {
+  var root = element.getRootNode ? element.getRootNode() : document.body;
```

Comment:
I think this should be

```js
return element.isConnected || document.body.contains(element);
```

... because either it has [isConnected](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected), which implies Shadow DOM support, or we can just check the old way.

Right now this would return true for being contained with a Shadow Root that's totally disconnected.

Reply:
Indeed, will change that.

Reply:
Done.